Provider crash during App Insights create

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.9.5

### AzureRM Provider Version

4.1.0

### Affected Resource(s)/Data Source(s)

azurerm_application_insights

### Terraform Configuration Files

```hcl
resource "azurerm_application_insights" "insights" {
  resource_group_name = "name"
  location            = "westus"
  name                = "insightsname"
  workspace_id        = "workspaceid"
  application_type    = "web"
  sampling_percentage = 0
  tags                = {
    environment = "qa"
    provisioner = "terraform"
    managed = true
  }
}
```


### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v4.1.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x4dbf169]

goroutine 233 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/applicationinsights.resourceApplicationInsightsCreateUpdate(0x0?, {0x729e040?, 0xc001fcf680})
        github.com/hashicorp/terraform-provider-azurerm/internal/services/applicationinsights/application_insights_resource.go:290 +0x1509
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x8b0c578?, {0x8b0c578?, 0xc004f90750?}, 0xd?, {0x729e040?, 0xc001fcf680?})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:766 +0x163
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000b8d500, {0x8b0c578, 0xc004f90750}, 0xc001b2f110, 0xc004c9ae00, {0x729e040, 0xc001fcf680})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/resource.go:909 +0xa89
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0008e19f8, {0x8b0c578?, 0xc004f90660?}, 0xc004f94000)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.33.0/helper/schema/grpc_provider.go:1078 +0xdbc
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x8b0c5b0?, {0x8b0c578?, 0xc004f90360?}, 0xc004f94000)
        github.com/hashicorp/terraform-plugin-mux@v0.15.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x193
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0xc00154afa0, {0x8b0c578?, 0xc004df9b30?}, 0xc0017d7810)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/tf5server/server.go:865 +0x3d0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x7fbde60?, 0xc00154afa0}, {0x8b0c578, 0xc004df9b30}, 0xc004c9ab80, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.23.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:518 +0x169
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000231000, {0x8b0c578, 0xc004df9aa0}, {0x8b38f60, 0xc000223c80}, 0xc004f88360, 0xc001668cf0, 0xe173798, 0x0)
        google.golang.org/grpc@v1.63.2/server.go:1369 +0xe23
google.golang.org/grpc.(*Server).handleStream(0xc000231000, {0x8b38f60, 0xc000223c80}, 0xc004f88360)
        google.golang.org/grpc@v1.63.2/server.go:1780 +0x1016
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.63.2/server.go:1019 +0x8b
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 39
        google.golang.org/grpc@v1.63.2/server.go:1030 +0x135

Error: The terraform-provider-azurerm_v4.1.0_x5 plugin crashed!
```


### Expected Behaviour

_No response_

### Actual Behaviour

_No response_

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/applicationinsights/application_insights_resource.go
+++ b/internal/services/applicationinsights/application_insights_resource.go
@@ -281,6 +281,11 @@ func resourceApplicationInsightsCreateUpdate(d *pluginsdk.ResourceData, meta int
 	if billingRead.Model == nil {
 		return fmt.Errorf("model is nil for billing features")
 	}
+
+	if billingRead.Model.DataVolumeCap == nil {
+		billingRead.Model.DataVolumeCap = &billing.ApplicationInsightsComponentDataVolumeCap{}
+	}
+
 	applicationInsightsComponentBillingFeatures := billing.ApplicationInsightsComponentBillingFeatures{
 		CurrentBillingFeatures: billingRead.Model.CurrentBillingFeatures,
 		DataVolumeCap:          billingRead.Model.DataVolumeCap,

